### PR TITLE
Fix Mold Breaker's interaction with Unaware

### PR DESCRIPTION
**Pull request by ascriptmaster**
It should compare this.activePokemon to "thing", which is the Pokemon that
is holding the Ability being processed, not "target".

Also fixed random typo where Mold Breaker might succeed even if the
ability were supressed.

### CODE REVIEW

File: test/simulator/moves/electricterrain.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ describe('Electric Terrain', function () {
 		battle = BattleEngine.Battle.construct();
 		battle.join('p1', 'Guest 1', 1, [{species: "Jolteon", ability: 'voltabsorb', moves: ['electricterrain', 'rest']}]);
 		battle.join('p2', 'Guest 2', 1, [{species: "Pidgeot", ability: 'keeneye', moves: ['doubleedge', 'rest']}]);
+		battle.seed = [0, 0, 0, 0];
```
Comment (Slayer95) [inline test/simulator/moves/electricterrain.js]: I assume a crit Double Edge could break this test otherwise?
We should use a weaker move instead in that case.

File: test/simulator/moves/thousandarrows.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ describe('Thousand Arrows', function () {
 		battle.commitDecisions();
 		assert.strictEqual(battle.p2.active[0].boosts.atk, 0);
 		assert.strictEqual(battle.p2.active[0].boosts.spa, 0);
+		battle.seed = [0, 0, 0, 0];
```
Comment (Slayer95) [inline test/simulator/moves/thousandarrows.js]: Ditto: increase EVs/IVs

Also, these should go in a different commit

File: test/simulator/moves/electricterrain.js (hunk 1 above)
Comment (ascriptmaster) [inline test/simulator/moves/electricterrain.js]: It's not even crit Double Edge. It's near-maxdamage crit Double Edge, with like ~35% of OHKO if it crits.

**Comment by ascriptmaster** [comment]
Blah, I'm just being uninspired. New commit and Shell Armor to block crits instead of seeding.

**Comment by ascriptmaster** [comment]
Also, you asked for Thick Fat tests, so here they are. Still all green.

**Comment by ascriptmaster** [comment]
And here is Multiscale. Good night everybody
